### ISSUE TITLE
[Adjust Rule] Fix SIM104 false positive inside async functions

### ISSUE THREAD

**Issue by AIGeneratedUsername** [MartinThoma/flake8-simplify#28]
## Desired change

* **Rule(s)**: SIM104
* **Adjustment**: do not raise a warning inside async functions

## Explanation

Python does not support 'yield from' inside async functions

https://www.python.org/dev/peps/pep-0525/#asynchronous-yield-from

If 'flake8-simplify' already can check if a function is "async" or not, then the issue has to be easy to fix. 

## Example

This is an example where the mentioned rule(s) would currently be suboptimal:

Copied from the https://www.python.org/dev/peps/pep-0525/#asynchronous-yield-from:
```
async def g2():
    async for v in g1():
        yield v

```

**Comment by MartinThoma** [comment]
Thank you for the example, @AIGeneratedUsername :

```
async def g2():
    async for v in g1():
        yield v
```

The AST looks like this:

```
        AsyncFunctionDef(
            name='g2',
            args=arguments(posonlyargs=[], args=[], vararg=None, kwonlyargs=[], kw_defaults=[], kwarg=None, defaults=[]),
            body=[
                AsyncFor(
                    target=Name(id='v', ctx=Store()),
                    iter=Call(
                        func=Name(id='g1', ctx=Load()),
                        args=[],
                        keywords=[],
                    ),
                    body=[
                        Expr(
                            value=Yield(
                                value=Name(id='v', ctx=Load()),
                            ),
                        ),
                    ],
                    orelse=[],
                    type_comment=None,
                ),
            ],
            decorator_list=[],
            returns=None,
            type_comment=None,
        ),
```

**Comment by MartinThoma** [comment]
Fixed via https://github.com/MartinThoma/flake8-simplify/pull/94
